### ISSUE TITLE
Support for transducer

### ISSUE THREAD

**Issue by gowtham1997** [Masao-Someki/espnet_onnx#17]
Thanks for creating this and simplifying onnx export. 

1. Do you have a timeline for the transducer ASR model support?

```python
from espnet2.bin.asr_inference import Speech2Text
import os
import yaml
from espnet_onnx.export import ModelExport
from pathlib import Path

m = ModelExport()

os.chdir("egs2/librispeech_100/asr1/")


transducer_conf = yaml.safe_load(Path('conf/decode_rnnt_conformer.yaml').read_text())

speech2text = Speech2Text(asr_train_config="exp/asr_train_rnnt_conformer_raw_en_bpe5000_sp/config.yaml",
                          asr_model_file="exp/asr_train_rnnt_conformer_raw_en_bpe5000_sp/latest.pth",
                          transducer_conf=transducer_conf["transducer_conf"],
                          lm_weight=0.0)

m.export(speech2text, 'onnx_export', quantize=True)
```
^ I was trying to export a sample transducer checkpoint like this and was running into attribute errors at `espnet_onnx/export/asr/export_asr.py" _create_config function`.  

2. Can you comment if the above snippet is correct? If I just provide None to attributes that dont exist (Eg: `model.beam_search`), would the conversion still work ?

**Comment by Masao-Someki** [comment]
@gowtham1997 
Thank you for using espnet_onnx!

> Do you have a timeline for the transducer ASR model support?

I usually develop espnet_onnx on a whim, so there is no schedule. But I will implement transducer-related scripts this weekend! I planned to implement some scripts for TTS, so I think I can publish transducer support with some TTS support.

> Can you comment if the above snippet is correct? 

Your snippet is almost correct. Currently, the transducer is not supported, so the `speech2text` model should be some other model.

> If I just provide None to attributes that don't exist (Eg: model.beam_search), would the conversion still work ?

Am I correct that you're saying that if someone can export an instance of `espnet2.bin.asr_inference.Speech2Text`, which does not have some attributes like `beam_search`?
If that's right, no, the conversion will not be completed.
Some attributes may not be required for conversion, but many attributes like encoder, decoder, and beam_search, and their configurations are required for inference of the onnx version.

**Comment by b-flo** [comment]
> Thank you for your information and suggestion! I think coding the transducer won't take that long, so I can code for it.
Please let me know when you merged the new transducer. I will support that new version then!

After discussion, the current Transducer version will stay so don't worry about working on it right now :)
If we add the new version, I'll try to take care of the export!

> About the onnx export, it will be very difficult if the structure of the model changes dynamically during inference. For this reason, I split the RNNDecoder into predecoder and decoder, where predecoder only contains attention models.
> So it would be better if the model does not change during inference, but there could be a way to implement the model in some way, so you don't have to care about exporting it!

Thanks for the info! Hum, it shouldn't be an issue then, I'll take a look.

**Comment by Masao-Someki** [comment]
Hi @gowtham1997, 

I just merged the #19 and now you can convert the transducer model!
It is not published on pypi yet, but you can clone and export your model.

**Comment by gowtham1997** [comment]
Thanks again.
Will test it out
